[WIP] Mixed precision (32-bit main) for dynamics & GEOS 11.4.2 feature set

FlorianDeconinck opened the pull request:
This is the full suite of changes to bring the dynamics to a 32-bit first (but implemented mixed) capabilities. It also brings non reconcilable changes to Remapping that where implemented in GEOS.

The entire codebase was ported from GEOS 11.4.2

gmao-ckung committed:
Added moist_cv_pt_last_step stencil and translate test

gmao-ckung committed:
Remove extraneous code from moist_cv_pt_last_step translate test

gmao-ckung committed:
Added cond_output stencil and corresponding translate test

gmao-ckung committed:
Updated remapping translate test with stencils needed for fv_mapz computation.  Note that since this isn't a parallel translate test, the mpp_global_sum function won't work, so it's commented out

FlorianDeconinck committed:
Merge branch 'fix/GEOS/TracerAdvection' into develop__GEOSv_11_4_2

FlorianDeconinck committed:
Specify correct dtype for `dpx`

FlorianDeconinck committed:
Merge branch 'fix/GEOSv11_4_2/DynCore' into develop__GEOSv_11_4_2

gmao-ckung committed:
Starting translate test with full j-axis being computed

gmao-ckung committed:
Modified remapping translate test to compute over entire domain.  Tested over first remapping j-loop

gmao-ckung committed:
Verified additional portions of remapping using full j.  Note that there may be an issue with computations using map1_ppm_u

gmao-ckung committed:
Verified mapz stencils within second j loop when executing with entire j range

gmao-ckung committed:
Removed extraneous code from remapping_GEOS translate test

gmao-ckung committed:
Converted GEOS remapping translate test into a parallel translate test.

gmao-ckung committed:
Debugging parallel remapping translate test

gmao-ckung committed:
Adding serialized fv_mpaz.F90.SER.v2

gmao-ckung committed:
Merge pull request #1 from gmao-ckung/feature/fv_mapz_GEOS_full_j_translate

Incorporating full j-axis computations and MPI testing into Remapping translate test

gmao-ckung committed:
Unified fv_mapz serialization file

FlorianDeconinck committed:
Update README

FlorianDeconinck committed:
Adapt XPPM translate test to serialization tweaks in Fortran

FlorianDeconinck committed:
Move `dpx` to be passed in the acoustics instead of temp

FlorianDeconinck committed:
`dpx` calculated on cell center

FlorianDeconinck committed:
Merge branch 'GEOS_update/yppm_xppm' into develop__GEOSv_11_4_2

FlorianDeconinck committed:
Merge branch 'fix/GEOS/D_SW' into develop__GEOSv_11_4_2

FlorianDeconinck committed:
Merge branch 'fix/GEOSv11_4_2/DynCore' into develop__GEOSv_11_4_2

FlorianDeconinck committed:
Add remapping reference to README

FlorianDeconinck committed:
Lint

FlorianDeconinck committed:
Remapping GEOS obejct (untested)

FlorianDeconinck committed:
Update dynamics with custom Remapping

FlorianDeconinck committed:
Typo

gmao-ckung committed:
Accounting for j=je+1 calculations in map u

gmao-ckung committed:
Corrected issue that caused pt to incorrected read netCDF file

gmao-ckung committed:
Continued verifying GEOS remapping/fv_mapz

gmao-ckung committed:
Adding fv_mapz serialization file

gmao-ckung committed:
Removed extraneous variables from GEOS remapping translate test and added fill condensate stencil to translate test

gmao-ckung committed:
Added fv_mapz serialization file

gmao-ckung committed:
Small correction in mpp_global_sum.py

FlorianDeconinck committed:
`remapper`: align with modified dynamics
`moist_cv`: lint

FlorianDeconinck committed:
Revert "Update dynamics with custom Remapping"

This reverts commit e90d0e7ae3de4b393313e70fa73e0bb6534e9245.

FlorianDeconinck committed:
Revert "Typo"

This reverts commit c39a2a447319a8c7f0fb5656668798402be23c1f.

FlorianDeconinck committed:
Merge branch 'feature/fv_mapz_GEOS' into develop__GEOSv_11_4_2

FlorianDeconinck committed:
Fluxes and courant number as independent parameters (out of state)

FlorianDeconinck committed:
Update translate test to independent fluxes/courant

FlorianDeconinck committed:
Add compute total energy function (partial)

FlorianDeconinck committed:
Fluxes & Courant number are local then increment global fields (f32)
Add DryMassRoudnOff code (f32)
Compute total energy in preamble
GEOS Lagrangian code

FlorianDeconinck committed:
Adapt translate test & allow for ULP<100 to pass test

FlorianDeconinck committed:
Merge branch 'fix/GEOSv11_4_2/DynCore' into develop__GEOSv_11_4_2

FlorianDeconinck committed:
Merge branch 'fix/GEOSv11_4_2/Dynamics' into develop__GEOSv_11_4_2

FlorianDeconinck committed:
Remove checkpointers

FlorianDeconinck committed:
Merge branch 'fix/GEOSv11_4_2/Dynamics' into develop__GEOSv_11_4_2

FlorianDeconinck committed:
Update README

FlorianDeconinck committed:
Fix `Tracers` on GPU

FlorianDeconinck committed:
Merge branch 'fix/GEOS/TracerAdvection' into develop__GEOSv_11_4_2

gmao-ckung committed:
Small edit to bounds in mpp_global_sum

gmao-ckung committed:
Removed extraneous code preventing fv_dynamics from executing

gmao-ckung committed:
Linting

gmao-ckung committed:
Enabled mpp_global_sum with flag to choose 'precise' or 'simplified' reduction calculation

gmao-ckung committed:
Merge pull request #2 from gmao-ckung/feature/fv_mapz_GEOS_update_Translate

Linted translate tests and updated mpp_global_sum with option to use simple MPI all_reduce call

gmao-ckung committed:
Small fixes to input and output dictionaries in translate test.  Not sure if these actually make a difference in the computation but it does providing more details about each serialized variable

gmao-ckung committed:
Using domain_compute in domain within stencil_factory calls

gmao-ckung committed:
Updated translate test with quantity factor to create temporary data

gmao-ckung committed:
Added updated fv_dynamics.F90.SER serialization file

gmao-ckung committed:
Made specific tracer parameter calls within remapping as calls from the tracer dictionary.  Note that the tracer key names right now need to match up with 'tracer_varaibles' from gt4py_utils.py

gmao-ckung committed:
Reorganizing code and accessing constants from PyFV3 and NDSL instead of serializing

gmao-ckung committed:
Code cleanup

gmao-ckung committed:
Laying groundwork to use translate test to test LangrangianToEulerian

gmao-ckung committed:
Continue development of LtoE translate test

gmao-ckung committed:
Continue L_to_E integration in translate remapping test

gmao-ckung committed:
Re-incorporated Tracer class into GEOS Remapping translate test

gmao-ckung committed:
updated fv_dynamics serialization file

FlorianDeconinck committed:
Fix Rayleigh Friction trigger configuration (raise)

FlorianDeconinck committed:
Merge branch 'fix/GEOSv11_4_2/Dynamics' into develop__GEOSv_11_4_2

gmao-ckung committed:
Added temporary calculation to flux calculation in fxadv.py that will reduce error rates of dpx values in D_SW

gmao-ckung committed:
Added fix to improve result matching of 'delpc' with GEOS codebase

gmao-ckung committed:
Added notes on porting

FlorianDeconinck committed:
Move damping increment to own K stencils
Trigger damping increment calculation exactly once

FlorianDeconinck committed:
Merge branch 'fix/RayleighDamping_mixed_precision' into develop__GEOSv_11_4_2

FlorianDeconinck committed:
Merge remote-tracking branch 'fdeconinck/fix/GEOS/D_SW' into develop__GEOSv_11_4_2

FlorianDeconinck committed:
Update to new NDSL API

FlorianDeconinck committed:
Merge branch 'fix/GEOSv11_4_2/Dynamics' into develop__GEOSv_11_4_2

FlorianDeconinck committed:
Fix `f32_correction` + lint

FlorianDeconinck committed:
Merge branch 'fix/GEOSv11_4_2/Dynamics' into develop__GEOSv_11_4_2

gmao-ckung committed:
Merge pull request #3 from gmao-ckung/feature/fv_mapz_GEOS_update_translate

Merging updated fv_mapz_GEOS translate test branch into fv_mapz_GEOS

FlorianDeconinck committed:
Fix inner power calculation with a mixed precision factor

FlorianDeconinck committed:
Merge branch 'fix/DelnFlux_f32_support' into develop__GEOSv_11_4_2

FlorianDeconinck committed:
Restore `cd` to it's original 64-bit factor

FlorianDeconinck committed:
Merge branch 'fix/GEOSv11_4_2/HyperDiffusionDamping' into develop__GEOSv_11_4_2

FlorianDeconinck committed:
Update README.md

FlorianDeconinck committed:
Restore `da_min` true 64-bit type

FlorianDeconinck committed:
Merge branch 'fix/GEOSv11_4_2/DynCore' into develop__GEOSv_11_4_2

FlorianDeconinck committed:
Remove debuggin artefacts

FlorianDeconinck committed:
Merge branch 'fix/DelnFlux_f32_support' into develop__GEOSv_11_4_2

FlorianDeconinck committed:
Fix `dd8` power function
Fix using proper precision for `da_min`
Fix: use proper `damp_tmp2` for `ke`

FlorianDeconinck committed:
Merge branch 'fix/GEOS/D_SW' into develop__GEOSv_11_4_2

FlorianDeconinck committed:
Match Fortran by multiplyin by 1/GRAV instead of divide

FlorianDeconinck committed:
Merge branch 'fix/GEOSv11_4_2/RiemanSolver' into develop__GEOSv_11_4_2

FlorianDeconinck committed:
Fix

FlorianDeconinck committed:
Undo `dt` refactoring in interpolation

FlorianDeconinck committed:
Adjust contravriant functions

FlorianDeconinck committed:
Docs

FlorianDeconinck committed:
Rename a2b_ord4 use for clarity

FlorianDeconinck committed:
Match Fortan circulation c grid operation factor
Do not inline sub-dt calculation

FlorianDeconinck committed:
Match fortran on cubic interpolation & edge interpolation
Fix order of op for UT update

FlorianDeconinck committed:
PK3 Halo match the fortran

FlorianDeconinck committed:
Make sure to use `dz_min` as parameters to allow for better GEOS integration

FlorianDeconinck committed:
Merge branch 'fix/GEOS/D_SW' into develop__GEOSv_11_4_2

FlorianDeconinck committed:
Merge branch 'fix/GEOSv11_4_2/DynCore' into develop__GEOSv_11_4_2

FlorianDeconinck committed:
Merge branch 'fix/GEOSv11_4_2/C_SW' into develop__GEOSv_11_4_2

FlorianDeconinck committed:
Merge branch 'fix/RayleighDamping_mixed_precision' into develop__GEOSv_11_4_2

FlorianDeconinck committed:
Merge branch 'fix/F32/nh_p_grad' into develop__GEOSv_11_4_2

FlorianDeconinck committed:
Match Fortran great circle distance
Fix doublu periodic order of op to match Fortran

FlorianDeconinck committed:
Merge branch 'fix/GEOSv11_4_2/A2B_Ord4' into develop__GEOSv_11_4_2

FlorianDeconinck committed:
Pass `dz_min` from `dyn_core` to the updater

FlorianDeconinck committed:
Merge branch 'fix/GEOSv11_4_2/DynCore' into develop__GEOSv_11_4_2

FlorianDeconinck committed:
Update `int` dtypes to precision capable ones

FlorianDeconinck committed:
Merge branch 'feature/fv_mapz_GEOS' into develop__GEOSv_11_4_2

FlorianDeconinck committed:
Fix remap after tracer merge

FlorianDeconinck committed:
Merge branch 'feature/fv_mapz_GEOS' into develop__GEOSv_11_4_2

FlorianDeconinck committed:
Proper type for da_min

FlorianDeconinck committed:
Merge branch 'fix/GEOSv11_4_2/Dynamics' into develop__GEOSv_11_4_2

FlorianDeconinck committed:
Fix bad merge

FlorianDeconinck committed:
Heat source from vorticity on the correct grid

FlorianDeconinck committed:
Insure abs on proper precision

FlorianDeconinck committed:
Merge branch 'fix/GEOS/D_SW' into develop__GEOSv_11_4_2

FlorianDeconinck committed:
Merge branch 'fix/GEOSv11_4_2/DynCore' into develop__GEOSv_11_4_2

FlorianDeconinck committed:
Add `dz_min` as a namelist input

FlorianDeconinck committed:
Merge branch 'fix/GEOSv11_4_2/Dynamics' into develop__GEOSv_11_4_2

FlorianDeconinck committed:
Hotfix for dyncore

FlorianDeconinck committed:
Add defaiult to `dz_min`

FlorianDeconinck committed:
Merge branch 'fix/GEOSv11_4_2/Dynamics' into develop__GEOSv_11_4_2

FlorianDeconinck committed:
Fix proper use of `gz` corner extension

FlorianDeconinck committed:
Match divg_d Fortran computation for divergence_corner

FlorianDeconinck committed:
Initialize `gz` to HUGE_R

FlorianDeconinck committed:
Merge branch 'fix/F32/UpdateDzC' into develop__GEOSv_11_4_2

FlorianDeconinck committed:
Merge branch 'fix/GEOSv11_4_2/C_SW' into develop__GEOSv_11_4_2

FlorianDeconinck committed:
Merge branch 'fix/GEOSv11_4_2/DynCore' into develop__GEOSv_11_4_2

FlorianDeconinck committed:
Fix translate test halo reach & pass `dz_min`

FlorianDeconinck committed:
Merge branch 'fix/F32/UpdateDzC' into develop__GEOSv_11_4_2

FlorianDeconinck committed:
Pass `dz_min` in translate test

FlorianDeconinck committed:
Merge branch 'fix/F32/UpdateDzD' into develop__GEOSv_11_4_2

FlorianDeconinck committed:
Clean up Translate test

FlorianDeconinck committed:
Merge branch 'feature/fv_mapz_GEOS' into develop__GEOSv_11_4_2

FlorianDeconinck committed:
Fix copy to cover interface dimensions

FlorianDeconinck committed:
Merge branch 'fix/GEOSv11_4_2/Dynamics' into develop__GEOSv_11_4_2

FlorianDeconinck committed:
Fix increment of fluxes for interface dims

FlorianDeconinck committed:
Merge branch 'fix/GEOSv11_4_2/Dynamics' into develop__GEOSv_11_4_2

FlorianDeconinck committed:
Fix set value to cover interface

FlorianDeconinck committed:
Flip refactor division to match fortran

FlorianDeconinck committed:
Localize ratio to match fortran

FlorianDeconinck committed:
Split p_grad by dimensions to forbid halos bleed

FlorianDeconinck committed:
Correct docs

FlorianDeconinck committed:
Restore `UnreachableStmtPruning` to fix dace:cpu layout transparency issue

FlorianDeconinck committed:
Extend translate test to halo

FlorianDeconinck committed:
Fix translate sizes

FlorianDeconinck committed:
Extend rieman solver test to halos

FlorianDeconinck committed:
Docs and verbose parameters

FlorianDeconinck committed:
Fix bad copy

FlorianDeconinck committed:
Fix precision of PressureAdjustedTemperature test

FlorianDeconinck committed:
Merge branch 'fix/GEOSv11_4_2/Dynamics' into develop__GEOSv_11_4_2

FlorianDeconinck committed:
Merge branch 'fix/F32/UpdateDzC' into develop__GEOSv_11_4_2

FlorianDeconinck committed:
Merge branch 'fix/GEOS/D_SW' into develop__GEOSv_11_4_2

FlorianDeconinck committed:
Merge branch 'fix/GEOSv11_4_2/A2B_Ord4' into develop__GEOSv_11_4_2

FlorianDeconinck committed:
Merge branch 'fix/F32/UpdateDzD' into develop__GEOSv_11_4_2

FlorianDeconinck committed:
Merge branch 'fix/GEOSv11_4_2/RiemanSolver' into develop__GEOSv_11_4_2

FlorianDeconinck committed:
Merge branch 'fix/DelnFlux_f32_support' into develop__GEOSv_11_4_2

FlorianDeconinck committed:
Merge branch 'fix/GEOS/TracerAdvection' into develop__GEOSv_11_4_2

FlorianDeconinck committed:
Merge branch 'fix/GEOSv11_4_2/DynCore' into develop__GEOSv_11_4_2

FlorianDeconinck committed:
Switch global reduce to `simplified` version

FlorianDeconinck committed:
Merge branch 'feature/fv_mapz_GEOS' into develop__GEOSv_11_4_2

FlorianDeconinck committed:
Reset `Tracers` for pyFV3 using `FieldBundle`

FlorianDeconinck committed:
Update tracers to `FieldBundle` and orchestration

FlorianDeconinck committed:
Merge branch 'feature/tracer_rework_part1' into develop__GEOSv_11_4_2

FlorianDeconinck committed:
Merge branch 'feature/fv_mapz_GEOS' into develop__GEOSv_11_4_2

FlorianDeconinck committed:
Update README.md

FlorianDeconinck committed:
Move MPP global sum to sub-modules

FlorianDeconinck committed:
Orchestrate MPP via callback

FlorianDeconinck committed:
Merge branch 'feature/fv_mapz_GEOS' into develop__GEOSv_11_4_2

FlorianDeconinck committed:
Fix imports on translate
Fix sum on 2 axis

FlorianDeconinck committed:
Bad merge

FlorianDeconinck committed:
Merge branch 'feature/fv_mapz_GEOS' into develop__GEOSv_11_4_2

FlorianDeconinck committed:
Move Tracers2D1L to FieldBundle tracers
Orchestrate

FlorianDeconinck committed:
Merge branch 'fix/GEOS/TracerAdvection' into develop__GEOSv_11_4_2

FlorianDeconinck committed:
Help `mypy` by going to `TypeAlias`

FlorianDeconinck committed:
Merge branch 'feature/tracer_rework_part1' into develop__GEOSv_11_4_2

FlorianDeconinck committed:
Fix liquid tracers order for GEOS

FlorianDeconinck committed:
Orchestrate Tracers

FlorianDeconinck committed:
Fix DycoreState for tracers, translate test
Remove workaround for Float(x) in orchestration

FlorianDeconinck committed:
Merge branch 'feature/tracer_rework_part1' into develop__GEOSv_11_4_2

FlorianDeconinck committed:
Merge branch 'fix/GEOS/TracerAdvection' into develop__GEOSv_11_4_2

FlorianDeconinck committed:
Merge branch 'fix/GEOSv11_4_2/DynCore' into develop__GEOSv_11_4_2

FlorianDeconinck committed:
Fix ComputeTotalEnergy w/ new Tracers

FlorianDeconinck committed:
Orchestrater compute total energy

FlorianDeconinck committed:
Orchestrate dynamics + new tracers structure

FlorianDeconinck committed:
Fix FVDynamics translate tests

FlorianDeconinck committed:
Orchestrate `ray_fast` w/ workaround for lazy call

FlorianDeconinck committed:
Fix assert in `sum`

FlorianDeconinck committed:
Remove redundant cast

FlorianDeconinck committed:
Update code to M2 standards

CharlesKrop committed:
Fixed UpdateDzC - computation now occurs at all needed points, no longer misses outermost x/y

CharlesKrop committed:
Brought back UpdateDzC documentation

gmao-ckung committed:
Fixed GlobalSum class and updated translate tests

FlorianDeconinck committed:
Update to NDSL 2025.10.00 API (Namelist, Corners)

FlorianDeconinck committed:
Merge remote-tracking branch 'NOAA/develop' into develop__GEOSv_11_4_2

FlorianDeconinck committed:
[WIP] Flak8 fixes

FlorianDeconinck committed:
[WIP] Non-tracers lint & fixes

FlorianDeconinck committed:
[WIP] Remove the DIM constants

FlorianDeconinck committed:
pyFV3 -> pyfv3

FlorianDeconinck committed:
[WIPP] Fix acoustics translate tests

FlorianDeconinck committed:
[WIP] Fix dyncore translate test (and some lint)

FlorianDeconinck committed:
Fix `dz_min` type in config
Fix tracers in Remap & Tracer2D1L modules

FlorianDeconinck committed:
Fix copy corners for numpy/quantity usage

FlorianDeconinck committed:
Merge branch 'update/modernize_remapping' into feature/tracers_to_ddims_field

FlorianDeconinck committed:
Introduce `DataDimensionFields` powered tracers
Rework `Remapping` code, translate tests and subcomponents to fit the new `tracers`

FlorianDeconinck committed:
Allow `copy_corners` to be distributed against all the memory type (and residency). This is to deal with the `no memoryview` feature of 3.12+

FlorianDeconinck committed:
Tracers 2D1L. WARNING: orchestration broken

FlorianDeconinck committed:
Update FVDynamics to new Tracers
Workaround for limited halo exchange behavior
Complete expected tracers for new FVDyn code
Clean up

FlorianDeconinck committed:
Merge remote-tracking branch 'NOAA/develop' into feature/tracers_to_ddims_field

FlorianDeconinck committed:
Fix orchestration for copy_corners_nord in delnflux

FlorianDeconinck committed:
Fix YPPM translate + clean up

FlorianDeconinck committed:
Fix standard override

FlorianDeconinck committed:
Remove no longer needed `norunroll`

FlorianDeconinck committed:
Fix translates

FlorianDeconinck committed:
Merge branch 'develop' into feature/tracers_to_ddims_field

FlorianDeconinck committed:
Use a global quantity of `Tracers`
Make tracers access via direct slicing (instead)
Fix `zero_qs` to be proper zero (use Quantity instad of Local)
Fix bad merge in `MapNTracers`

FlorianDeconinck committed:
Deactivate broken DryConvectiveAdjustment

FlorianDeconinck committed:
[TMP] Link to specific branch for `ndsl`

FlorianDeconinck committed:
Lint

FlorianDeconinck committed:
Rework tracers to allow re-entry. Clean up `Fillz` accessors

FlorianDeconinck committed:
Lint

FlorianDeconinck committed:
Lint

FlorianDeconinck committed:
Fix doubly periodic test

FlorianDeconinck committed:
More `.data` cleanup

FlorianDeconinck committed:
External trigger for pyFV3 CI pulls this branch

FlorianDeconinck committed:
Pull on `DataDimensionField` branch for `pySHieLD`

FlorianDeconinck committed:
Pace CI to branch hook

FlorianDeconinck committed:
Update `pySHiELD` branch

FlorianDeconinck committed:
Merge branch 'develop' into develop__GEOSv_11_4_2

FlorianDeconinck committed:
Merge branch 'feature/tracers_to_ddims_field' into develop__GEOSv_11_4_2

FlorianDeconinck committed:
Use FVTracers + fix merge

FlorianDeconinck committed:
Fix TracerCMax test and temporary alloc

FlorianDeconinck committed:
Tracer2D1L: restore code ate by merge

FlorianDeconinck committed:
Verbose TracerCMax callback issue w/ ticket

FlorianDeconinck committed:
Update to `develop` NDSL

FlorianDeconinck committed:
FVDynamics orchestration

FlorianDeconinck committed:
Fix tracers marshalling in translate_RemappingGEOS

FlorianDeconinck committed:
Move NDSLRuntime across Acoustics codes

FlorianDeconinck committed:
Condig: pass `dz_min` down from a namelist read
Initialize the all of `zs` with `phis`, including halos, as does the Fortran

FlorianDeconinck committed:
Move threshold of 100 ULP for all backends on FV

FlorianDeconinck committed:
Mixed precision for the Configuration

FlorianDeconinck committed:
NDSRuntime the XPPM & YPPM

FlorianDeconinck committed:
Bad merge brought a dual application of the `pt ` to potential density `pt`

FlorianDeconinck committed:
Use `da_min` and `da_min_c` directly

FlorianDeconinck committed:
Lint

FlorianDeconinck committed:
Remove archived old fv_mapz

FlorianDeconinck committed:
Implementing `nwat==0`

FlorianDeconinck committed:
Move timing collection into it's own file to calm down mypy

FlorianDeconinck committed:
Align xppm and yppm

FlorianDeconinck committed:
Align xppm & yppm again

FlorianDeconinck committed:
Upgrade all to NDSLRuntime

FlorianDeconinck committed:
To locals: D_SW

FlorianDeconinck committed:
To locals: DynCore

FlorianDeconinck committed:
To locals: Remapping_GEOS + Validation thresholds

FlorianDeconinck committed:
To locals: Tracer2D1L

FlorianDeconinck committed:
Refactor unused local or local that can be inlined in stencils as temporaries

FlorianDeconinck committed:
To Locals: FVDynamics
- pass `wsd` directly in Acoustics instead of a weird init level cache
- one unexplianed failure to go Local with `te0_2d`

FlorianDeconinck committed:
Remove dead code + lint on Tracers

FlorianDeconinck committed:
Workaround in copy corners for SDFG -> STree bug

twicki committed:
corner fix for orchestration

FlorianDeconinck committed:
Guard for `nwat<6`

oelbert commented on pyfv3/_config.py:
I'm relieved this doesn't break type checking anymore

FlorianDeconinck committed:
Fix type of Tracers in compute total energy

FlorianDeconinck committed:
Restore temparay fields in tracer adjustement code

twicki committed:
Merge branch 'develop__GEOSv_11_4_2' of github.com:NOAA-GFDL/PyFV3 into develop__GEOSv_11_4_2

twicki committed:
working stree corners

twicki committed:
map fix

twicki committed:
Merge commit 'a15246b7552f4019b318db6130e6884604d62158' into develop__GEOSv_11_4_2

romanc committed:
fix linting

romanc committed:
cleanup of old/unused copy_corner codes

romanc committed:
cleanup of old/unused corner codes

FlorianDeconinck committed:
Remove unused `boilerplate` need

FlorianDeconinck committed:
Swap `set_value` for custom reset to go around bad schedule tree merge behavior

FlorianDeconinck committed:
Fix `copy_corner`: take topology into account

romanc committed:
unrelated: fix typo

romanc committed:
remove warning about uninitialized transients

romanc committed:
clean up translate test of Tracer2D1L

FlorianDeconinck committed:
Remove logging - we have `-g` now for crashes

romanc committed:
cleanup warnings in DelnFlux translate test

romanc committed:
tests: reduce warnings in fvtp2d translate test

romanc committed:
tests: cleanup Del6VtFlux translate test

romanc committed:
tests: cleanups in A2B_Ord4 translate test

romanc committed:
a2b_ord4: use integer powers in great_circle_dist

romanc committed:
a2b_ord4: fix interpolation in doubly_periodic_a2b_ord4()

Fix regression introduced in

https://github.com/NOAA-GFDL/pyFV3/commit/b494884a29a61849a91eb4d0af9de3758f83f1ca

romanc committed:
acoustics translate test: avoid .data

FlorianDeconinck committed:
Fix corner copy distribution based on topology

Commit:
Fix for floating point exponents

Commit:
Fix for floating point exponents

FlorianDeconinck committed:
Workaround to not have `damping_increment` be a view

Commit:
More floating point exponent fixes

FlorianDeconinck committed:
Merge remote-tracking branch 'Katrina/develop__GEOSv_11_4_2' into develop__GEOSv_11_4_2

FlorianDeconinck committed:
RayFast: swap locals to quantity because of lazy init

CharlesKrop committed:
removing a local that causes a validation issue for pt in DynCore

CharlesKrop committed:
linting

FlorianDeconinck committed:
Add a version `AGrid2BGridFourthOrder` with inplace modification of input.
Introduce a `local` into the original a2b_ord4
Block merging with callbacks in NH_P_Grad to go around a bug

FlorianDeconinck committed:
Rework `neg_adj3` to go around an issue with schedule tree (unsqueezed memlet)

FlorianDeconinck committed:
Move `qmin` to a module default to remove cast on-the-fly

FlorianDeconinck committed:
Remove `optional` scalar that trips GPU code generation

romanc committed:
Minor: cleanup comments in a2b_ord4

romanc committed:
minor: consistently reference tracer by external "name"

romanc committed:
remove unsued debug variable from updatedzc

romanc committed:
cleanup translate test of rayleigh damping

romanc committed:
updatedzd: change float constant 1 -> 1.0

romanc committed:
fix: dont use `.data` in del2cubed (fixes test)

This makes the translate test (`orch:dace:gpu:IJK`) go from numerical
failures in ~8% to completely green.

romanc committed:
delnflux: don't use .data in DelnFluxNoSG

FlorianDeconinck committed:
Custom `OptimizationConfig` for FVDynamics

FlorianDeconinck committed:
Better K schedule for compute total energy

FlorianDeconinck committed:
Centralize OptimizationConfig

FlorianDeconinck committed:
FV uses central optimization config

FlorianDeconinck committed:
Fix compute_total_energy vertical loop schedule

FlorianDeconinck committed:
Deactivate all schedule tree on FV

FlorianDeconinck committed:
Kernalize in GPU

FlorianDeconinck committed:
Deactivate all tree on FV (kernalize succeed but bridge fails)

romanc committed:
fix: run copy corners on GPU

romanc committed:
Revert "fix: run copy corners on GPU"

This reverts commit b7937d9170d06b2cf7bf086d42a5a6e41817599e.

romanc committed:
no else after return in if

romanc committed:
rename: `kernalize` -> `kernelize` in optimization config

romanc committed:
ci: pull current working branch of NDSL

romanc committed:
Merge remote-tracking branch 'origin/develop' into develop__GEOSv_11_4_2

romanc committed:
tests: minor cleanup

romanc committed:
tests: make XTP_U a standalone test (not inheriting from YTP_V)

romanc committed:
minor cleanups in tests

romanc committed:
fix: No (early!) return in DelnFlux

romanc committed:
do not return from C_SW

Looks like this was just added for test convenience, but since we have
the C_SW object in the test as well, we can just grab `delpc` and `ptc`
from there.

romanc committed:
cleanup: marking / removing unused code

FlorianDeconinck committed:
Merge branch 'develop' into develop__GEOSv_11_4_2

FlorianDeconinck committed:
Merge branch 'develop' into develop__GEOSv_11_4_2

FlorianDeconinck committed:
Merge remote-tracking branch 'origin/develop' into develop__GEOSv_11_4_2

FlorianDeconinck committed:
Use K axis name that helps the schedule tree

FlorianDeconinck committed:
Update to unreleased 2026.08.X

romanc committed:
Merge remote-tracking branch 'origin/develop' into develop__GEOSv_11_4_2
